halo2_proofs/src/circuit.rs

```diff
@@ -441,6 +443,11 @@ pub trait Layouter<F: Field> {
         row: usize,
     ) -> Result<(), Error>;
 
+    /// Queries the value of the given challenge.
+    ///
+    /// Returns `Value::unknown()` if the current synthesis phase is before the challenge can be queried.
+    fn get_challenge(&mut self, challenge: Challenge) -> Value<F>;
```

Comment:
I don't see any reason for this to be mutable.
```suggestion
    fn get_challenge(&self, challenge: Challenge) -> Value<F>;
```